### PR TITLE
fix inference file

### PR DESCRIPTION
**Pull request by hadiyouness10**
I fixed the inference for mmdetection pvt training. It now saves the result into a specific file, and then we access it to send the data to the label studio UI. 

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: custom/inference.py [hunk 1]
```diff
@@ -46,30 +46,63 @@ def print_fps(self):
...
+    parser.add_argument('--config_file', type=str, help='Path to model config file')
+    parser.add_argument('--checkpoint_file', type=str, help='Path to checkpoint file')
+    parser.add_argument('--inf_dir', type=str, help='Directory containing images for inference')
+    parser.add_argument('--inf_out_dir', type=str, help='Directory to save inference results')
+    parser.add_argument('--threshold', type=str, help='Directory to save inference results')
+    parser.add_argument('--output_file', type=str, help='Directory to save inference results')
+
+    args = parser.parse_args()
+    if not args.threshold:
+        args.threshold = 50
```
Comment (christian-holland) [inline custom/inference.py]: this is a bit confusing, the more common convention is to use thresholds between 0 and 1, so I would not add the factor of 100 here and parse a `type=float` instead.

File: custom/inference.py [hunk 2]
```diff
@@ -46,30 +46,63 @@ def print_fps(self):
...
+    parser = argparse.ArgumentParser(description='Run inference on images')
+    parser.add_argument('--config_file', type=str, help='Path to model config file')
+    parser.add_argument('--checkpoint_file', type=str, help='Path to checkpoint file')
+    parser.add_argument('--inf_dir', type=str, help='Directory containing images for inference')
+    parser.add_argument('--inf_out_dir', type=str, help='Directory to save inference results')
+    parser.add_argument('--threshold', type=str, help='Directory to save inference results')
+    parser.add_argument('--output_file', type=str, help='Directory to save inference results')
+
+    args = parser.parse_args()
+    if not args.threshold:
```
Comment (christian-holland) [inline custom/inference.py]: instead of this, you can use the "default" argument of `add_argument`

File: custom/inference.py [hunk 3]
```diff
@@ -46,30 +46,63 @@ def print_fps(self):
...
+    def run(self, threshold = 50):
+            results = []
+            for imn in os.listdir(self.inf_dir):
+                img = f'{self.inf_dir}/{imn}'
+                self.fps_logger.start_record()
+                result = inference_detector(self.model, img)
+                results.append({img : extract_bounding_boxes(result, threshold)})
+                self.fps_logger.end_record()
+                # model.show_result(img, result)
+                # self.model.show_result(img, result, out_file=f'{self.inf_out_dir}/{imn}', score_thr=0.5)
```
Comment (christian-holland) [inline custom/inference.py]: we should keep the option to visualize the images. You can add an optional argument `output_image_dir` that, if given, writes the images there.

File: custom/inference.py [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
-
+import argparse
 # Specify the path to model config and checkpoint file
 # config_file = 'configs/custom/my_retinanet_pvt-t_fpn_1x_coco.py'
 # checkpoint_file = 'work_dirs/my_retinanet_pvt-t_fpn_1x_coco/epoch_7.pth'
 
-config_file = 'data/work_dirs/fstb0/old_logs/512_384_24ep_rot90/fstb0.py'
-checkpoint_file = 'data/work_dirs/fstb0/old_logs/512_384_24ep_rot90/epoch_24.pth'
+# config_file = '/data/label_studio_datasets/forklift_person_id_84/ml/models/2023_06_06_17_05_37.016432/auto_config.py'
+# checkpoint_file = '/data/label_studio_datasets/forklift_person_id_84/ml/models/2023_06_06_17_05_37.016432/latest.pth'
```
Comment (christian-holland) [inline custom/inference.py]:
```suggestion
```

File: custom/inference.py [hunk 5]
```diff
@@ -21,8 +21,8 @@
 # inf_dir = 'data/fst/random_select_/random_select'
 # inf_out_dir = 'data/fst/random_select_/fstv2_first'
 
-inf_dir = "data/fst/random_select_/test_img/2"
-inf_out_dir = 'data/fst/random_select_/pytorch_rot90'
+# inf_dir = "/data/label_studio_datasets/forklift_person_id_84/images_test"
+# inf_out_dir = '/data/label_studio_datasets/forklift_person_id_84/inference_result_annotations'
```
Comment (christian-holland) [inline custom/inference.py]:
```suggestion
```
looks like these are not needed anymore, can be removed

File: custom/inference.py (hunk 1 above)
Comment (hadiyouness10) [reply custom/inference.py]: noted will do

File: custom/inference.py (hunk 3 above)
Comment (hadiyouness10) [reply custom/inference.py]: we already have the inf_out_dir, i will use that

**Comment by hadiyouness10** [comment]
do we have precommits in this repo? @christian-holland

File: custom/inference.py [hunk 6]
```diff
@@ -46,30 +40,61 @@ def print_fps(self):
...
-    def run(self):
-        for imn in os.listdir(self.inf_dir):
-            img = f'{inf_dir}/{imn}'
-            self.fps_logger.start_record()
-            result = inference_detector(self.model, img)
-            self.fps_logger.end_record()
-            # model.show_result(img, result)
-            self.model.show_result(img, result, out_file=f'{self.inf_out_dir}/{imn}', score_thr=0.5)
+    
+    def run(self, threshold = 0.5):
```
Comment (christian-holland) [inline custom/inference.py]:
```suggestion
    def run(self, threshold=0.5):
```
there are no precommits repo, but you can use the auto formatter of your IDE for some basic formatting (for example for pycharm ctrl + a then ctrl + alt + L)

**Review by christian-holland — APPROVED** [review]

**Pull request merged**